hal/inc/concurrent_hal.h

```diff
@@ -179,6 +179,7 @@ int os_mutex_trylock(os_mutex_t mutex);
 int os_mutex_unlock(os_mutex_t mutex);
 
 int os_mutex_recursive_create(os_mutex_recursive_t* mutex);
+int os_mutex_recursive_create_static(os_mutex_recursive_t* mutex);
```

Comment:
Yes, that was my intention initially. I'd like to see static variants for `os_primitive_create` as well in the future, but for the intents and purposes of this PR, adding them here seems like an overkill.